src/application/components/CodeBlock.tsx

```diff
@@ -0,0 +1,182 @@
+import { ColorTheme, colorTheme } from "../theme";
+import { CopyIcon } from "./icons/Copy";
+import { Button } from "./Button";
+
+interface SyntaxHighlighterProps {
+  className?: string;
+  language: Language;
+  code: string;
+}
+
+const { code } = colors.tokens;
+
+const getTheme = (mode: ColorTheme): PrismTheme => {
+  const isDark = mode === ColorTheme.Dark;
+
+  return {
+    plain: {
+      color: "currentColor",
+    },
+    styles: [
+      {
+        types: ["important", "bold"],
+        style: {
+          fontWeight: "bold",
+        },
+      },
+      {
+        types: ["italic"],
+        style: {
+          fontStyle: "italic",
+        },
+      },
+      {
+        types: ["entity"],
+        style: {
+          cursor: "help",
+        },
+      },
+      {
+        types: ["selector", "attr-name", "char", "builtin", "inserted"],
+        style: {
+          color: isDark ? code.d.dark : code.d.base,
+        },
+      },
+      {
+        types: ["comment", "prolog", "doctype", "cdata"],
+        style: {
+          color: isDark ? code.b.dark : code.b.base,
+          fontWeight: "bold",
+        },
+      },
+      {
+        types: ["scalar", "attr-value"],
+        style: {
+          color: isDark ? code.c.dark : code.c.base,
+        },
+      },
+      {
+        types: ["url", "attr-value"],
+        style: {
+          color: isDark ? code.g.dark : code.g.base,
+        },
+      },
+      {
+        types: ["punctuation", "constant"],
+        style: {
+          color: isDark ? code.a.dark : code.a.base,
+        },
+      },
+      {
+        types: ["property", "tag", "boolean", "number", "symbol"],
+        style: {
+          color: isDark ? code.c.dark : code.c.base,
+        },
+      },
+      {
+        types: ["class-name"],
+        style: {
+          color: isDark ? code.a.dark : code.a.base,
+        },
+      },
+      {
+        types: ["property", "function"],
+        style: {
+          color: isDark ? code.d.dark : code.d.base,
+        },
+      },
+      {
+        types: ["atrule", "attr-value", "keyword"],
+        style: {
+          color: "inherit",
+          background: "transparent",
+        },
+      },
+      {
+        types: ["atrule", "attr-value", "keyword"],
+        style: {
+          color: isDark ? code.b.dark : code.b.base,
+        },
+      },
+      {
+        types: ["string"],
+        style: {
+          color: isDark ? code.g.dark : code.g.base,
+        },
+      },
+      {
+        types: [
+          "regex",
+          "important",
+          "variable",
+          "description",
+          "comment",
+          "prolog",
+          "doctype",
+          "cdata",
+        ],
+        style: {
+          color: isDark ? code.f.dark : code.f.base,
+        },
+      },
+    ],
+  };
+};
+
+export const CodeBlock = ({
+  code,
+  className,
+  language,
+}: SyntaxHighlighterProps) => {
+  const activeTheme = getTheme(useReactiveVar(colorTheme));
```

Comment:
```suggestion
  const colorThemeValue = useReactiveVar(colorTheme)
  const activeTheme = useMemo(() => getTheme(colorThemeValue), [colorThemeValue]);
```

As this creates a new object on every call, it might make sense to keep the reference stable.

Reply:
Good call. Done! [44ce295](https://github.com/apollographql/apollo-client-devtools/pull/1180/commits/44ce2955759267a728c0e17fc94e8ba51b18204b)